crates/wasmparser/benches/benchmark.rs

```diff
@@ -119,70 +193,39 @@ fn collect_benchmark_inputs() -> Vec<BenchmarkInput> {
 fn it_works_benchmark(c: &mut Criterion) {
     let mut inputs = collect_benchmark_inputs();
     // Filter out all benchmark inputs that fail to parse via `wasmparser`.
-    inputs.retain(|input| {
-        let mut parser = Parser::new(input.wasm.as_slice());
-        'outer: loop {
-            match parser.read() {
-                ParserState::Error(_) => break 'outer false,
-                ParserState::EndWasm => break 'outer true,
-                _ => continue,
-            }
-        }
-    });
+    inputs.retain(|input| read_all_wasm(input.wasm.as_slice()).is_ok());
     c.bench_function("it works benchmark", move |b| {
         b.iter(|| {
             for input in &mut inputs {
-                let _ = black_box(read_all_wasm(
-                    &input.path,
-                    Parser::new(input.wasm.as_slice()),
-                ));
+                read_all_wasm(input.wasm.as_slice()).unwrap();
             }
         })
     });
 }
 
-fn validator_not_fails_benchmark(c: &mut Criterion) {
-    let mut inputs = collect_benchmark_inputs();
-    // Filter out all benchmark inputs that fail to validate via `wasmparser`.
-    inputs.retain(|input| {
-        let mut parser = ValidatingParser::new(input.wasm.as_slice(), VALIDATOR_CONFIG);
-        'outer: loop {
-            match parser.read() {
-                ParserState::Error(_) => break 'outer false,
-                ParserState::EndWasm => break 'outer true,
-                _ => continue,
-            }
-        }
-    });
-    c.bench_function("validator no fails benchmark", move |b| {
-        b.iter(|| {
-            for input in &mut inputs {
-                let _ = black_box(read_all_wasm(
-                    &input.path,
-                    ValidatingParser::new(input.wasm.as_slice(), VALIDATOR_CONFIG),
-                ));
-            }
-        });
-    });
-}
-
 fn validate_benchmark(c: &mut Criterion) {
+    fn validator() -> Validator {
+        let mut ret = Validator::new();
+        ret.wasm_reference_types(true)
+            .wasm_multi_value(true)
+            .wasm_simd(true)
+            .wasm_module_linking(true)
+            .wasm_bulk_memory(true)
+            .wasm_threads(true)
+            .wasm_tail_call(true);
+        return ret;
```

Comment:
Couldn't you chain it because it has `&mut self` receivers? If so: If this is a common way to instantiate `Validator` instances we might prefer a builder pattern here.

Reply:
My personally style is to use `&mut self -> &mut self` for builder style rather than `self -> self`. This is a downside that you have to bind a local variable every now and then, though.

Reply:
Both works, yeah. Whatever you prefer. The advantage with `self -> self` is that you can more easily use type-level to protect against wrong usage which is why I generally prefer it but not really needed here.